### PR TITLE
[internal] Simplify import resolvers

### LINKED ISSUE #1707: [BUG]: Linting raises library-not-found problem when import `type_extensions`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

We raise a library-not-found problem when import type_extentions

```
if TYPE_CHECKING:
    from typing_extensions import ParamSpec

    P = ParamSpec("P")
```

### Expected Behavior

No problem should be raised. Maybe we should white list this library

### Steps To Reproduce

1. Create unit test for above code snippet

### Cloud

Azure

### Operating System

macOS

### Version

latest via Databricks CLI

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Remove chain of import resolvers in favor of a single resolver.

Supersedes #1709 

Resolves #1707

### CODE REVIEW

File: src/databricks/labs/ucx/source_code/graph.py [hunk 1]
```diff
@@ -277,21 +277,9 @@ def _fail(code: str, message: str) -> MaybeDependency:
 
 class BaseImportResolver(abc.ABC):
```
Comment (nfx) [inline src/databricks/labs/ucx/source_code/graph.py:278]: redundant: we no longer need `BaseImportResolver`

File: src/databricks/labs/ucx/source_code/files.py [hunk 2]
```diff
@@ -111,13 +112,11 @@ def __repr__(self):
 
 class LocalFileResolver(BaseImportResolver, BaseFileResolver):
```
Comment (nfx) [inline src/databricks/labs/ucx/source_code/files.py:113]:
```suggestion
class ImportResolver(BaseFileResolver):
```
and get rid of BaseImportResolver.

File: src/databricks/labs/ucx/source_code/whitelist.py [hunk 3]
```diff
@@ -22,16 +20,13 @@
...
+        super().__init__()
         self._whitelist = whitelist
 
-    def with_next_resolver(self, resolver: BaseImportResolver) -> BaseImportResolver:
-        return WhitelistResolver(self._whitelist, resolver)
-
-    def resolve_import(self, path_lookup: PathLookup, name: str) -> MaybeDependency:
+    def resolve_import(self, name: str) -> MaybeDependency | None:
         # TODO attach compatibility to dependency, see https://github.com/databrickslabs/ucx/issues/1382
         compatibility = self._whitelist.compatibility(name)
```
Comment (nfx) [inline src/databricks/labs/ucx/source_code/whitelist.py:31]: move the body of this method as private method to `ImportResolver` class and get rid of `WhitelistResolver`.

**Review by nfx — CHANGES_REQUESTED** [review]
simplify even further.

File: src/databricks/labs/ucx/source_code/whitelist.py (hunk 3 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/whitelist.py:31]: done

File: src/databricks/labs/ucx/source_code/files.py (hunk 2 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/files.py:113]: renamed to ImportFileResolver

**Pull request merged**